### PR TITLE
Remove SDK 0.9.4 idioms

### LINKED ISSUE #1575: Adapt to the new implementation of platform SDK 0.15.0
- [x] Remove `pureAccountSigningMetaFor` from `SigMetadataLookup` (https://github.com/hashgraph/hedera-services/pull/1573#discussion_r649663692 and https://github.com/hashgraph/hedera-services/pull/1573#discussion_r649666833)
- [x] Do not call replace() after getForModify() in platform SDK 0.15.0 (https://github.com/hashgraph/hedera-services/pull/1573#discussion_r649665086)

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #1575

**Summary of the change**:
- Remove unnecessary pure/impure account lookups pattern.
- Remove unnecessary calls to `FCM#replace` after `FCM#getForModify`.
- Standardize on superclass `throwIfImmutable()` in `MerkleAccountTokens` mutators.

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/BackedAccountLookup.java [hunk 1]
```diff
@@ -36,19 +36,10 @@ public BackedAccountLookup(BackingStore<AccountID, MerkleAccount> accounts) {
...
-	public SafeLookupResult<AccountSigningMetadata> pureSafeLookup(AccountID id) {
-		return lookup(id, true);
-	}
-
-	private SafeLookupResult<AccountSigningMetadata> lookup(AccountID id, boolean usePure) {
 		if (!accounts.contains(id)) {
 			return SafeLookupResult.failure(MISSING_ACCOUNT);
 		}
-		var account = usePure ? accounts.getUnsafeRef(id) : accounts.getRef(id);
+		var account = accounts.getUnsafeRef(id);
```
Comment (qnswirlds) [inline hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/BackedAccountLookup.java:42]: nit: Should we rename the function as we are call `getUnsafeRef`?

File: hedera-node/src/test/java/com/hedera/services/store/AccountStoreTest.java [hunk 2]
```diff
@@ -125,7 +124,7 @@ void persistenceUpdatesTokens() {
 		subject.persistAccount(model);
 
 		// then:
-		verify(accounts).replace(miscMerkleId, expectedReplacement);
+		assertEquals(expectedReplacement, miscMerkleAccount);
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/store/AccountStoreTest.java:127]: I think we should keep this `verify` but use `never()`

File: hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java [hunk 3]
```diff
@@ -111,7 +111,7 @@ void persistsExtantTokenRelAsExpected() {
 		subject.persistTokenRelationship(modelTokenRel);
 
 		// then:
-		verify(tokenRels).replace(miscTokenRelId, expectedReplacementTokenRel);
+		assertEquals(expectedReplacementTokenRel, miscTokenMerkleRel);
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:114]: Same here

File: hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java [hunk 4]
```diff
@@ -202,7 +202,7 @@ void savesTokenAsExpected() {
 		subject.persistToken(modelToken);
 
 		// then:
-		verify(tokens).replace(merkleTokenId, expectedReplacementToken);
+		assertEquals(expectedReplacementToken, merkleToken);
```
Comment (qnswirlds) [inline hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:205]: And here

**Review by qnswirlds — APPROVED** [review]
Overall LGTM. Just a few places need clarified.

File: hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/BackedAccountLookup.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/sigs/metadata/lookups/BackedAccountLookup.java]: Changed to `getImmutableRef()` 👍

File: hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:115]: Added `verify(..., never())...`

File: hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java (hunk 4 above)
Comment (tinker-michaelj) [reply hedera-node/src/test/java/com/hedera/services/store/TypedTokenStoreTest.java:207]: Added `verify(..., never())...`

**Review by qnswirlds — APPROVED** [review]

**Pull request merged**